rooms.js

```diff
@@ -933,16 +933,34 @@ class BattleRoom extends Room {
 	}
 	requestKickInactive(user, force) {
 		if (this.resetTimer) {
-			if (user) this.sendUser(user, '|inactive|The inactivity timer is already counting down.');
-			return false;
+			let timerSetByUser = user && (this.resetUser === user.userid || (user in this.game.players && this.resetUser === '+'));
```

Comment:
...I hope you aren't implicitly asking me to move this to another file altogether. I'd try, sure, but...